libp2p/Host.cpp

```diff
@@ -328,21 +289,17 @@ void Host::runAcceptor()
 	{
 		clog(NetConnect) << "Listening on local port " << m_listenPort << " (public: " << m_tcpPublic << ")";
 		m_accepting = true;
-		m_socket.reset(new bi::tcp::socket(m_ioService));
-		m_tcp4Acceptor.async_accept(*m_socket, [=](boost::system::error_code ec)
+		
+		bi::tcp::socket* s = new bi::tcp::socket(m_ioService);
+		m_tcp4Acceptor.async_accept(*s, [=](boost::system::error_code ec)
```

Comment:
Is the lambda guaranteed to be called? Otherwise we leak a socket. Perhaps use a unique_ptr? In this case, it gets deallocated even if the lambda is not called (and only destroyed).